### PR TITLE
Feat/action table --run-build

### PR DESCRIPTION
**Pull request by jokomyn**
changelog:
- integrate move item
- integrate move table
- adjustment response and payload flow order
- fix bug member

### CODE REVIEW

**Comment by akarintitech** [comment]
# SonarQube Code Analytics 
## Quality Gate Status: OK

passed

[Sonar report](https://sonar.akarinti.tech/dashboard?id=dayacore_pos_web&pullRequest=24) (https://sonar.akarinti.tech/dashboard?id=dayacore_pos_web&pullRequest=24)

## Additional information
*The following metrics might not affect the Quality Gate status but improving them will improve your project code quality.*

## Issues
- Bugs: security_a [0 Bugs](https://sonar.akarinti.tech/project/issues?id=dayacore_pos_web&resolved=false&sinceLeakPeriod=true&types=BUG&pullRequest=24)

- Vulnerabilities: security_a [0 Vulnerabilities](https://sonar.akarinti.tech/project/issues?id=dayacore_pos_web&resolved=false&sinceLeakPeriod=true&types=VULNERABILITY&pullRequest=24)

- Code Smells: security_a [0 Code Smells](https://sonar.akarinti.tech/project/issues?id=dayacore_pos_web&resolved=false&sinceLeakPeriod=true&types=CODE_SMELL&pullRequest=24)

- Security Hotspots: security_a [0 Security Hotspots](https://sonar.akarinti.tech/security_hotspots?id=dayacore_pos_web&pullRequest=24&inNewCodePeriod=true)



## Coverage and Duplications
- Coverage:  [0.00% Coverage](https://sonar.akarinti.tech/project/issues?id=dayacore_pos_web&metric=new_coverage&view=list&pullRequest=24)

- Duplications:  [0.00% Duplication](https://sonar.akarinti.tech/project/issues?id=dayacore_pos_web&metric=new_duplicated_lines_density&view=list&pullRequest=24)

File: src/app/components/Features/MenuList/MenuList.component.tsx [hunk 1]
```diff
@@ -149,52 +149,92 @@ const MenuList: React.FC<menuProps> = ({data='', detailBook='', onChangeAddInfo,
...
-      const response = JSON.parse(lastMessage.data)
-      if(response.response_schema.response_code === '200') {
-        setListMenu(stateMenu === 'ITEM_MENU' ? response.response_output?.detail : response.response_output?.list?.content)
+  // useEffect(() => {
+  //   if (lastMessage !== null && checkJson(lastMessage.data)) {
+  //     setMessageHistory((prev) => prev.concat(lastMessage));
+  //     setListMenu([])
+  //     const response = JSON.parse(lastMessage.data)
+  //     if(response.response_schema.response_code === '200') {
+  //       setListMenu(stateMenu === 'ITEM_MENU' ? response.response_output?.detail : response.response_output?.list?.content)
```
Comment (aldiz1711) [inline src/app/components/Features/MenuList/MenuList.component.tsx:158]: jika sudah tidak dipakai tolong di remove

File: src/app/components/Features/MenuList/MenuList.component.tsx [hunk 2]
```diff
@@ -149,52 +149,92 @@ const MenuList: React.FC<menuProps> = ({data='', detailBook='', onChangeAddInfo,
...
+  //         getMenuById(salesModeId)
+  //         break;
+  //       case 'CATEGORY_DETAIL':
+  //         getCategoryDetail(salesModeId)
+  //         break;
+  //       default:
+  //         getCategory(salesModeId)
+  //         break;
+  //     }
+  //   }
```
Comment (aldiz1711) [inline src/app/components/Features/MenuList/MenuList.component.tsx:177]: jika sudah tidak dipakai tolong di remove

File: src/app/components/Features/MenuList/MenuList.component.tsx [hunk 3]
```diff
@@ -149,52 +149,92 @@ const MenuList: React.FC<menuProps> = ({data='', detailBook='', onChangeAddInfo,
...
+  };
+
+  const getMenuById = (salesModeId: number) => {
+    // sendMessage(JSON.stringify({ type: "POST", data: { menu_category_detail_id: selectedDetailCategory?.id, sales_mode_id: salesModeId } }))
+  
+    actionGetDataMenu('menu', { type: "POST", data: { menu_category_detail_id: selectedDetailCategory?.id, sales_mode_id: salesModeId } })
+  };
+
+  const getCategoryDetail = (salesModeId: number) => {
+    // sendMessage(JSON.stringify({ type: "POST", data: { menu_category_id: selectedCategory?.id, sales_mode_id: salesModeId } }))
```
Comment (aldiz1711) [inline src/app/components/Features/MenuList/MenuList.component.tsx:206]: jika sudah tidak dipakai tolong di remove

**Comment by akarintitech** [comment]
# SonarQube Code Analytics 
## Quality Gate Status: OK

passed

[Sonar report](https://sonar.akarinti.tech/dashboard?id=dayacore_pos_web&pullRequest=24) (https://sonar.akarinti.tech/dashboard?id=dayacore_pos_web&pullRequest=24)

## Additional information
*The following metrics might not affect the Quality Gate status but improving them will improve your project code quality.*

## Issues
- Bugs: security_a [0 Bugs](https://sonar.akarinti.tech/project/issues?id=dayacore_pos_web&resolved=false&sinceLeakPeriod=true&types=BUG&pullRequest=24)

- Vulnerabilities: security_a [0 Vulnerabilities](https://sonar.akarinti.tech/project/issues?id=dayacore_pos_web&resolved=false&sinceLeakPeriod=true&types=VULNERABILITY&pullRequest=24)

- Code Smells: security_a [0 Code Smells](https://sonar.akarinti.tech/project/issues?id=dayacore_pos_web&resolved=false&sinceLeakPeriod=true&types=CODE_SMELL&pullRequest=24)

- Security Hotspots: security_a [0 Security Hotspots](https://sonar.akarinti.tech/security_hotspots?id=dayacore_pos_web&pullRequest=24&inNewCodePeriod=true)



## Coverage and Duplications
- Coverage:  [0.00% Coverage](https://sonar.akarinti.tech/project/issues?id=dayacore_pos_web&metric=new_coverage&view=list&pullRequest=24)

- Duplications:  [0.00% Duplication](https://sonar.akarinti.tech/project/issues?id=dayacore_pos_web&metric=new_duplicated_lines_density&view=list&pullRequest=24)

**Pull request merged**